### PR TITLE
fix(TBD-9299): update Databricks distribution constants

### PR DESCRIPTION
**Pull request by sponomarova**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+// You should have received a copy of the agreement
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.hadoop.distribution.constants.databricks;
+
+@SuppressWarnings("nls")
+public interface IDatabricks350Distribution extends IDatabricksDistribution {
```
Comment (amedvedenko) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java]: is it a project requirement to add `I` before the name of the interface? As far as I know, this is a bad practice. IF it's a project requirement - then it's ok to me)

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+	 public final static String VERSION = "Databricks_3_5";
+
+	 public static final String VERSION_DISPLAY = "3.5 LTS (includes Apache Spark 2.2.1, Scala 2.11)";
+}
+
+
```
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java]: I can't see the purpose of externalising the version in a separate interface knowing that this interface will be used by only one concrete class DBR350Distribution

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java (hunk 1 above)
Comment (lbourgeois) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java]: I can't see the purpose of externalising the version in a separate interface knowing that this interface will be used by only one concrete class DBR350Distribution

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks540Distribution.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//
+// You should have received a copy of the agreement
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.hadoop.distribution.constants.databricks;
+
+@SuppressWarnings("nls")
+public interface IDatabricks540Distribution extends IDatabricksDistribution {
```
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks540Distribution.java]: Same remark as BRD350

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java (hunk 1 above)
Comment (sponomarova) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/IDatabricks350Distribution.java]: Agree. There is no wide use of this interface.
Will remove it.

**Comment by sponomarova** [comment]
TBD-9299 fix was modified.
No more need in these additional changes.

**Pull request closed without merging**